### DISCUSSION TITLE
meta: why is neovide a text editor and not a terminal emulator?

### DISCUSSION THREAD

**Discussion by auroraanna** [neovide/neovide#2495] [General]
i showed a friend neovide and they wondered why you didn't make it a terminal instead of a text editor, i.e. a terminal with smooth scrolling and the neovide cursor animations. then you could open regular neovim in a terminal and have the same smooth scrolling and cursor animations but you get to have them just in the terminal too, when you're not editing text.

**Comment by auroraanna** [comment]
is there technical reasons why this wouldn't be possible?

**Comment by Kethku** [comment]
Not all of the features in neovide map to a terminal grid. Window shadows, animated window positioning, and smooth scrolling within a nvim window for instance would be very difficult to implement correctly with just a terminal emulator.

It would absolutely be possible to create a terminal emulator that uses neovide's front end, but the work just doesn't feel worth the squeeze to me personally. That said we've discussed many times the idea of refactoring neovide to make the front end more usable for other projects. If that could be done, I think a terminal emulator would be an awesome project to build with it along with front ends for other guis like helix or whatever else.

TLDR: Sure. It could be done. To do it would be a lot of work that if somebody was sufficiently motivated to do and wants to dive in for, I'd personally love to help out with. Nobody has done that yet though.

(gonna convert this to a discussion as its not really a feature request and has been discussed before)

**Reply by ikimmit** [reply]
Helix boosted with Neovide's features would be simply awesome.

**Comment by Azarattum** [comment]
I also think that neovide has a potential to be a great general purpose terminal emulator. These effects are so satisfying, it would be cool to be able to have them outside of neovim.
